metagraph/core/compiler.py

```diff
@@ -67,6 +73,7 @@ def _note_subgraph(chain):
             and len(key_dependents) == 1
             and next_key in key_dependents
             and key in next_key_dependencies
+            and key not in output_keys  # output keys must be at the end of a chain
```

Comment:
Since this is intended to check if `next_key` should be inserted into the current chain, shouldn't this be `and next_key not in output_keys` ?

Reply:
This is preventing a chain from getting longer once it contains an output key (`key` is the last item in the chain).  We can add an output key to the end of a chain as long as we add no more.  Checking `next_key` would make an output chain stop one node too soon, I think.